model/src/main/java/hu/blackbelt/judo/meta/jsl/runtime/TypeInfo.java

```diff
@@ -293,6 +293,29 @@ public boolean isCompatible(TypeInfo other) {
 		return this.baseType == other.baseType;
 	}
 
+	public boolean isCompatibleCollection(TypeInfo other) {
+		if (this.baseType == BaseType.UNDEFINED || other.baseType == BaseType.UNDEFINED) {
+			return false;
+		}
+
+		if (!other.isCollection()) {
+			return false;
+		}
+		
+		if (this.isDeclaration() || other.isDeclaration()) {
+			return false;
+		}
+		
+		if (this.baseType == BaseType.ENTITY && other.baseType == BaseType.ENTITY)
+		{
+			return modelExtension.isEqual(this.type, other.type)
+					|| modelExtension.getSuperEntityTypes((EntityDeclaration)other.type).stream()
+						.anyMatch(e -> modelExtension.isEqual(e, this.type));
+		} else {
+			return false;
+		}
```

Comment:
These if-else-s could be merged into one and I prefer using _Object#equals_ instead of `==`

Reply:
If-else-s are deliberately not merged for better understanding, because this method is a comparison method that performs the comparison step by step.